docs/source/pages/Algorithm.rst

```diff
@@ -1,47 +1,121 @@
+|diagram|
+
+.. _select-atoms:
+
+1. Select atoms of reference
+-----------------------------
+
+The first step in the algorithm consists in selecting atoms that will be used as
+a reference to derive a surface. This selection will be contained in an
+AtomGroup. Typically in biological membranes, lipid headgroups are the most
+common elements to use as an AtomGroup of reference. 
+
+|atoms|
+
+.. _set-grid:
+
+2. Set grid
+-------------
+The dimensions of the grid are determined by the size of the simulation box
+contained in the `MDAnalysis Universe`_. The grid comprises `n_x_bins` x `n_y_bins` number of bins.
 
 |grid|
 
-Once the grid is populated with the atoms in the group of reference, the associated
-`z` coordinate of each atom is stored in an array assigned to each [i, j] unit cell.
-The length of these arrays will depend on the number of atoms of reference that
-are mapped to a given `[i, j]` unit cell. Subsequently, once all values of `z` are
-stored, the mean value of `z` stored in that array is calculated. Since this operation 
-is performed for every frame, the length of the array containing the mean `z` position
-will be determined by the number of frames iterated in the trajectory.
+For every atom in the AtomGroup of reference, MembraneCurvature assigns an index
+in the grid, according to their respective `x` and `y` coordinates. i.e. `(x, y) ↦ [l, m]`. 
+
+.. note::
+  Unless the user provides a different input, the dimensions of the grid will be determined by
 
-.. code-block:: python
+  .. code-block:: python
 
-    len(reference_atoms[i,j]) == len(u.trajectory)
+      grid_dimension_x = (0, universe.dimensions[0])
+      grid_dimension_y = (0, universe.dimensions[1])
 
-After iterating over every frame of the trajectory, the mean value of the stored `z`
-coordinate are calculated. These mean values of `z` are the ones used to derive the
-surface.
+Once the grid is populated according to the coordinates of the atoms in the
+AtomGroup of reference, the associated `z` coordinate of each atom in the AtomGroup 
+is stored in an array assigned to each `[l, m]` index.
 
-With the defined surface, then values of mean (`H`) and Gaussian (`K`) curvature
-are calculated. This analysis returns a 2-dimensional array of `m x n` with the values of mean
-mean and Gaussian curvature. The length of this array is `m x n`.
+.. _derive-surface-curvature:
 
-More information on how to visualize the results of the MDAnalysis Membrane 
-Curvature toolkit can be found in the Visualization_ page .
+3. Derive surface and calculate curvature
+------------------------------------------
 
-.. |grid| image:: ../_static/gridmap.png
+Once the surface formed by the atoms of reference is derived, values of mean (`H`)
+and Gaussian (`K`) curvature are calculated according to their respective equations.
+
+For every frame of the trajectory, the surface derived from the `AtomGroup` is
+calculated and stored in :attr:`MembraneCurvature.results.z_surface`.
```

Comment:
```suggestion
calculated and stored in :attr:`MembraneCurvature.results.z_surface<membrane_curvature.base.MembraneCurvature.results.z_surface>`.
```

I can't click this; you may need to fully qualify the path. I haven't checked the above code but something like that should link to "membrane_curvature.base.MembraneCurvature.results.z_surface", but only show "MembraneCurvature.results.z_surface" to the reader.

Reply:
Bruh... this I couldn't fix :( 
I tried:
- :attr:`MembraneCurvature.results.z_surface<membrane_curvature.base.MembraneCurvature.results.z_surface>`
- :attr:`MembraneCurvature.results.z_surface<MembraneCurvature.results.z_surface>`
- :attr:`~MembraneCurvature.results.average_z_surface<membrane_curvature.base.MembraneCurvature.results.average_z_surface>`

None of them worked. Honestly, I can't identify what's the issue here.

Reply:
Sorry, I think this was a brainfart on my part -- I think it makes sense that you can't make this clickable because `.z_surface` is not an attribute of MembraneCurvature but of the Results object, if that makes sense. My apologies!